sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveTables.scala

```diff
@@ -0,0 +1,190 @@
+        TableChange.updateColumnComment(colName.toArray, newComment)
+      }
+      AlterTable(catalog, tblName.toIdentifier, table, typeChange.toSeq ++ commentChange)
+
+    case AlterTableRenameColumnStatement(
+         CatalogAndTable(catalog, tblName, Left(v1Table)), col, newName) =>
+      throw new AnalysisException("RENAME COLUMN is not supported with v1 table.")
+
+    case AlterTableRenameColumnStatement(
+         CatalogAndTable(catalog, tblName, Right(table)), col, newName) =>
+      val changes = Seq(TableChange.renameColumn(col.toArray, newName))
+      AlterTable(catalog, tblName.toIdentifier, table, changes)
+
+    case AlterTableDropColumnsStatement(
+         CatalogAndTable(catalog, tblName, Left(v1Table)), cols) =>
+      throw new AnalysisException("DROP COLUMN is not supported with v1 table.")
+
+    case AlterTableDropColumnsStatement(
+         CatalogAndTable(catalog, tblName, Right(table)), cols) =>
+      val changes = cols.map(col => TableChange.deleteColumn(col.toArray))
+      AlterTable(catalog, tblName.toIdentifier, table, changes)
+
+    case AlterTableSetPropertiesStatement(
+         CatalogAndTable(catalog, tblName, Left(v1Table)), props) =>
+      AlterTableSetPropertiesCommand(tblName.toV1Identifier, props, isView = false)
+
+    case AlterTableSetPropertiesStatement(
+         CatalogAndTable(catalog, tblName, Right(table)), props) =>
+      val changes = props.map { case (key, value) =>
+        TableChange.setProperty(key, value)
+      }
+      AlterTable(catalog.asTableCatalog, tblName.toIdentifier, table, changes.toSeq)
+
+    case AlterTableUnsetPropertiesStatement(
+         CatalogAndTable(catalog, tblName, Left(v1Table)), keys, ifExists) =>
+      AlterTableUnsetPropertiesCommand(tblName.toV1Identifier, keys, ifExists, isView = false)
+
+    // TODO: v2 `UNSET TBLPROPERTIES` should respect the ifExists flag.
+    case AlterTableUnsetPropertiesStatement(
+         CatalogAndTable(catalog, tblName, Right(table)), keys, _) =>
+      val changes = keys.map(key => TableChange.removeProperty(key))
+      AlterTable(catalog, tblName.toIdentifier, table, changes)
+
+    case AlterTableSetLocationStatement(
+         CatalogAndTable(catalog, tblName, Left(v1Table)), newLoc) =>
+      AlterTableSetLocationCommand(tblName.toV1Identifier, None, newLoc)
+
+    case AlterTableSetLocationStatement(
+         CatalogAndTable(catalog, tblName, Right(table)), newLoc) =>
+      val changes = Seq(TableChange.setProperty("location", newLoc))
+      AlterTable(catalog.asTableCatalog, tblName.toIdentifier, table, changes)
+
+    // The v1 `AlterTableSetLocationCommand` throws NuSuchTable exception at runtime, while the v2
+    // command throws AnalysisException at analysis time. Here we convert to the v1 command to keep
+    // the exception type unchanged.
+    // TODO: unify the table not found exception for all ALTER TABLE commands.
+    case AlterTableSetLocationStatement(tblName, newLoc) =>
```

Comment:
This is a general problem that, v1 ALTER TABLE commands do table lookup at runtime and throw `NuSuchTable` exception, while v2 ALTER TABLE commands do table lookup at analysis time and throw `AnalysisException`.

I think the v2 behavior is more reasonable, but I'd like to avoid behavior changes in this refactor PR.